docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,197 @@
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT product_version_id
+        FROM product_versions
+        WHERE
+            product_name=%s AND
+            version_string=%s
+    """, (product_name, version_string))
+    ret = cursor.fetchone()
+    if ret is None:
+        return None
+    return ret[0]
+
+
+def insert_adu(conn, adu_count, adu_date, product_name, os_name, version_string):
+    """Inserts ADU data into product_adu table"""
+    # Fetch the product_version_id we need
+    product_version_id = fetch_product_version_id(conn, product_name, version_string)
+
+    if product_version_id is None:
+        print('No product_version_id for %s %s' % (product_name, version_string))
+        return
+
+    # Insert the data
+    cursor = conn.cursor()
+    try:
+        cursor.execute("""
+            INSERT INTO product_adu (adu_count, adu_date, os_name, product_version_id)
+            VALUES (%s, %s, %s, %s)
+        """, (adu_count, adu_date, os_name, product_version_id))
+    except psycopg2.IntegrityError as exc:
+        # Hitting an IntegrityError here probably means this data is in the table already, so we
+        # just skip it.
+        pass
+    conn.commit()
+
+
+def get_connection():
+    """Builds a connection using ConnectionContext environment variables
+
+    :returns: postgres connection
+
+    """
+    # This uses the same environment variables and defaults as postgres ConnectionContext
+    host = os.environ.get('resource.postgresql.database_hostname', 'localhost')
+    port = int(os.environ.get('resource.postgresql.database_port', '5432'))
+    username = os.environ.get('secrets.postgresql.database_username', 'breakpad_rw')
+    password = os.environ.get('secrets.postgresql.database_password', 'aPassword')
+    dbname = os.environ.get('resource.postgresql.database_name', 'breakpad')
+
+    local_config = {
+        'database_hostname': host,
+        'database_port': port,
+        'database_username': username,
+        'database_password': password,
+        'database_name': dbname
+    }
+    dsn = (
+        "host=%(database_hostname)s "
+        "dbname=%(database_name)s "
+        "port=%(database_port)s "
+        "user=%(database_username)s "
+        "password=%(database_password)s"
+    ) % local_config
+    return psycopg2.connect(dsn)
+
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--products', default='Firefox',
+        help='comma separated set of products to fetch ADI for'
+    )
+    parser.add_argument(
+        '--start_date', default='LASTWEEK', help='start date (mm/dd/yyyy)'
+    )
+    parser.add_argument(
+        '--end_date', default='TODAY', help='end date (mm/dd/yyyy)'
+    )
+    parser.add_argument(
+        '--endpoint', default='https://crash-stats.mozilla.com/api/ADI/',
+        help='endpoint for ADI API'
+    )
+    args = parser.parse_args()
+
+    today = datetime.datetime.now()
```

Comment:
Yeah, it's cryptic. In the footer of the webapp it says "All dates are UTC" and we have some code somewhere that makes sure the database's timezone is always set in UTC.